astropy/modeling/functional_models.py

```diff
@@ -572,6 +572,92 @@ def fit_deriv(x, amplitude, x_0, fwhm):
         return [d_amplitude, d_x_0, d_fwhm]
 
 
+class Voigt1D(Fittable1DModel):
+    """
+    One dimensional model for the Voigt profile.
+
+    Parameters
+    ----------
+    pos : float
+        Position of the peak
+    amp_L : float
+        The Lorentzian amplitude
+    fwhm_L : float
+        The Lorentzian full width at half maximum
+    fwhm_G : float
+        The Gaussian full width at half maximum
+
+    See Also
+    --------
+    Gaussian1D, Lorentz1D
+
+    Notes
+    -----
+    Algorithm for the computation taken from
+    McLean, A. B., Mitchell, C. E. J. & Swanston, D. M. Implementation of an
+    efficient analytical approximation to the Voigt function for photoemission
+    lineshape analysis. Journal of Electron Spectroscopy and Related Phenomena
+    69, 125-132 (1994)
+    """
+
+    pos = Parameter()
+    amp_L = Parameter()
+    fwhm_L = Parameter()
+    fwhm_G = Parameter()
+
+    @staticmethod
+    def evaluate(x, pos, amp_L, fwhm_L, fwhm_G):
+        A = [-1.2150, -1.3509, -1.2150, -1.3509]
+        B = [1.2359, 0.3786, -1.2359, -0.3786]
+        C = [-0.3085, 0.5906, 0.3085, 0.5906]
+        D = [0.0210, -1.1858, -0.0210, 1.1858]
+
+        sqrt_ln2 = np.sqrt(np.log(2))
+        sqrt_pi = np.sqrt(np.pi)
+        X = (x - pos) * 2 * sqrt_ln2 / fwhm_G
+        Y = fwhm_L * sqrt_ln2 / fwhm_G
+        V = 0
+
+        for i in range(4):
+            V += (C[i] * (Y - A[i]) + D[i] * (X - B[i]))/(((Y - A[i]) ** 2 + (X - B[i]) ** 2))
+
```

Comment:
Simplifying to ```V = np.sum(C * (Y - A) + D * (X - B))/(((Y - A) ** 2 + (X - B) ** 2))``` isn't quite working because arrays ```X``` and ```A, B, C, D``` are of different sizes..